### PR TITLE
[MM-T662] Add e2e test for /join command

### PR DESCRIPTION
**Pull request by seongwon-kang**
#### Summary

Adds test case for `/join` command:
* Privileged user try to `/join` to private channel with channel name
* Privileged user try to `/join` to private channel with tilde(~) + channel name

* Unprivileged user try to `/join` to private channel with channel name
* Unprivileged user try to `/join` to private channel with tilde(~) + channel name

#### Ticket Link
Github Issue: Resolves https://github.com/mattermost/mattermost-server/issues/15698
JIRA Ticket: MM-T662

### CODE REVIEW

**Review by larkox — APPROVED** [review]
LGTM. Thanks!

File: e2e/cypress/integration/integrations/slash_commands_spec.js [hunk 1]
```diff
@@ -55,6 +55,51 @@ describe('Integrations', () => {
         });
     });
 
+    it('MM-T662 /join command for private channels', () => {
+        const privateChannelName = user1.username;
```
Comment (saturninoabril) [inline e2e/cypress/integration/integrations/slash_commands_spec.js:59]: nit: Please use name other than the username. Could be,
```suggestion
        const privateChannelName = 'private-channel';
```

File: e2e/cypress/integration/integrations/slash_commands_spec.js [hunk 2]
```diff
@@ -55,6 +55,51 @@ describe('Integrations', () => {
...
+        cy.apiLogin(user1);
+        cy.visit(testChannelUrl1);
+
+        // # User 1 Create a private channel, with ${channelName}
+        cy.get('#createPrivateChannel').click();
+        cy.get('#private').click();
+        cy.get('#newChannelName').type(privateChannelName);
+        cy.get('#submitNewChannel').click();
+
+        // # User with privilege try join command without tilde
```
Comment (saturninoabril) [inline e2e/cypress/integration/integrations/slash_commands_spec.js:70]:
```suggestion
        // # User with privilege try /join command without tilde
```

File: e2e/cypress/integration/integrations/slash_commands_spec.js [hunk 3]
```diff
@@ -55,6 +55,51 @@ describe('Integrations', () => {
...
+        cy.get('#publicChannelList').findByText('Town Square').click();
+        cy.postMessage(`/join ${privateChannelName}`);
+
+        // * private channel should be active
```
Comment (saturninoabril) [inline e2e/cypress/integration/integrations/slash_commands_spec.js:74]: nit: Please capitalize comment
```suggestion
        // * Private channel should be active
```

File: e2e/cypress/integration/integrations/slash_commands_spec.js [hunk 4]
```diff
@@ -55,6 +55,51 @@ describe('Integrations', () => {
...
+        cy.get('#privateChannelList').get('.active').should('contain', privateChannelName);
+
+        // # User with privilege try join command with tilde
```
Comment (saturninoabril) [inline e2e/cypress/integration/integrations/slash_commands_spec.js:77]:
```suggestion
        // # User with privilege try /join command with tilde
```

File: e2e/cypress/integration/integrations/slash_commands_spec.js [hunk 5]
```diff
@@ -55,6 +55,51 @@ describe('Integrations', () => {
...
+        cy.postMessage(`/join ~${privateChannelName}`);
+
+        // * private channel should be active
+        cy.get('#privateChannelList').get('.active').should('contain', privateChannelName);
+
+        // # Login with user without privilege
+        cy.apiLogin(user2);
+        cy.visit(testChannelUrl1);
+
+        // # User without privilege try join command without tilde
```
Comment (saturninoabril) [inline e2e/cypress/integration/integrations/slash_commands_spec.js:88]:
```suggestion
        // # User without privilege try /join command without tilde
```

File: e2e/cypress/integration/integrations/slash_commands_spec.js [hunk 6]
```diff
@@ -55,6 +55,51 @@ describe('Integrations', () => {
...
+        cy.get('#publicChannelList').findByText('Town Square').click();
+        cy.postMessage(`/join ${privateChannelName}`);
+
+        // * Error message should be presented.
+        cy.getLastPost().should('contain', 'An error occurred while joining the channel.').and('contain', 'System');
+
+        // # User without privilege try join command with tilde
```
Comment (saturninoabril) [inline e2e/cypress/integration/integrations/slash_commands_spec.js:95]:
```suggestion
        // # User without privilege try /join command with tilde
```

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Thanks @seongwon-kang and welcome! Looks good to me except for minor suggestions.

File: e2e/cypress/integration/integrations/slash_commands_spec.js (hunk 6 above)
Comment (seongwon-kang) [reply e2e/cypress/integration/integrations/slash_commands_spec.js]: Thanks for review, @saturninoabril!

**Comment by seongwon-kang** [comment]
Thanks for review @saturninoabril!

**Review by saturninoabril — APPROVED** [review]
Thanks @seongwon-kang! Looks good to me.

**Pull request merged**